src/ansys/fluent/core/workflow.py

```diff
@@ -11,6 +12,27 @@
 from ansys.fluent.core.data_model_cache import DataModelCache
 from ansys.fluent.core.services.datamodel_se import PyCallableStateObject, PyMenuGeneric
 
+AttrMap = {}
+
+
+def camel_to_snake_case(input_arg: str) -> str:
+    """Convert camel case input string to snake case output string."""
+    return re.sub("((?<=[a-z0-9])[A-Z]|(?!^)[A-Z](?=[a-z]))", r"_\1", input_arg).lower()
+
+
+def populate_and_use_snake_case(attr: str, attr_dict):
+    """Populate the snake-case attribute map and return snake case of the passed
+    attribute."""
+    if attr.islower():
+        for key in attr_dict.keys():
+            if camel_to_snake_case(str(key)) not in AttrMap:
+                AttrMap[camel_to_snake_case(str(key))] = str(key)
```

Comment:
- This doesn't keep the result of `cache_to_snake_case()` which would avoid an unnecessary second call.
- The functionality is not specific to workflow.